### DISCUSSION TITLE
textureSampleLevel vs textureLoad

### DISCUSSION THREAD

**Discussion by redcamel** [gpuweb/gpuweb#3694] [Q&A] [answered]
textureSampleLevel vs textureLoad

I am researching post image processing through compute shaders.

I saw in the wgsl description that textureLoad takes a single texel without sampling.
My expectation is that textureLoad is faster than textureSampleLevel, but I want to know if this is the correct interpretation.

**Comment by PENGUINLIONG** [comment, marked as answer]
Basically the same on most GPUs because texture operations are accelerated by hardware texture units. Just ensure you use `texture_2d` rather than `texture_storage_2d` so you don't miss the hardware accelerator.

**Reply by kainino0x** [reply]
I don't know the answer but I think I agree with this. The actual memory access probably dwarfs any extra work done by using a (non-filtering) sampler. Choose the one that is convenient for you. If you want sampling, use a sampler, if you don't, then don't.
